cdap-api/src/main/java/co/cask/cdap/api/data/batch/Input.java

```diff
@@ -0,0 +1,223 @@
+   *
+   * @param datasetName the name of the input dataset
+   * @param splits the data selection splits. If null, will use the splits defined by the dataset.
+   */
+  public static Input ofDataset(String datasetName, @Nullable Iterable<? extends Split> splits) {
+    return ofDataset(datasetName, RuntimeArguments.NO_ARGUMENTS, splits);
+  }
+
+  /**
+   * Returns an Input defined by a dataset.
+   *
+   * @param datasetName the name of the input dataset
+   * @param arguments the arguments to use when instantiating the dataset
+   * @param splits the data selection splits. If null, will use the splits defined by the dataset.
+   */
+  public static Input ofDataset(String datasetName, Map<String, String> arguments,
+                                             Iterable<? extends Split> splits) {
+    return new DatasetInput(datasetName, arguments, splits);
+  }
+
+  /**
+   * Returns an Input defined by an InputFormatProvider.
+   *
+   * @param inputName the name of the input
+   */
+  public static Input of(String inputName, InputFormatProvider inputFormatProvider) {
+    return new InputFormatProviderInput(inputName, inputFormatProvider);
+  }
+
+  /**
+   * Returns an Input defined with the given stream name with all time range.
+   *
+   * @param streamName Name of the stream.
+   */
+  public static Input ofStream(String streamName) {
+    return ofStream(new StreamBatchReadable(streamName, 0, Long.MAX_VALUE));
+  }
+
+  /**
+   * Returns an Input defined by a stream with the given properties.
+   *
+   * @param streamName Name of the stream.
+   * @param startTime Start timestamp in milliseconds.
+   * @param endTime End timestamp in milliseconds.
+   */
+  public static Input ofStream(String streamName, long startTime, long endTime) {
+    return ofStream(new StreamBatchReadable(streamName, startTime, endTime));
+  }
+
+  /**
+   * Returns an Input defined by a stream with the given properties.
+   *
+   * @param streamName Name of the stream
+   * @param startTime Start timestamp in milliseconds (inclusive) of stream events provided to the job
+   * @param endTime End timestamp in milliseconds (exclusive) of stream events provided to the job
+   * @param decoderType The {@link StreamEventDecoder} class for decoding {@link StreamEvent}
+   */
+  public static Input ofStream(String streamName, long startTime,
+                             long endTime, Class<? extends StreamEventDecoder> decoderType) {
+    return ofStream(new StreamBatchReadable(streamName, startTime, endTime, decoderType));
+  }
+
+  /**
+   * Returns an Input defined by a stream with the given properties.
+   *
+   * @param streamName Name of the stream
+   * @param startTime Start timestamp in milliseconds (inclusive) of stream events provided to the job
+   * @param endTime End timestamp in milliseconds (exclusive) of stream events provided to the job
+   * @param bodyFormatSpec The {@link FormatSpecification} class for decoding {@link StreamEvent}
+   */
+  @Beta
```

Comment:
Only this method with these parameters annotated Beta in StreamBatchReadable.
I think its because body FormatSpecification API is Beta.